sklearn/linear_model/_base.py

```diff
@@ -542,9 +543,45 @@ def rmatvec(b):
                     for j in range(y.shape[1]))
                 self.coef_ = np.vstack([out[0] for out in outs])
                 self._residues = np.vstack([out[3] for out in outs])
+        elif (X.shape[1] <= 2
+              and np.all(np.linalg.eigvals(np.dot(X, np.transpose(X))) > 0)):
```

Comment:
It's not clear to me where this comes from. Also, this call in itself is probably more expensive than doing the actual regression. I think as a first iteration I would let the user pick the solver and not have a heuristic.